Tools/WebGPUPlayground/AppPlayground/WebGPUCoordinator.m

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "WebGPUCoordinator.h"
+
+@import Foundation;
+@import WebGPU;
+
+#define UNUSED_PARAM(variable) (void)variable
+
+@interface Scheduler: NSObject
+- (instancetype)init;
+- (void)schedule:(WGPUWorkItem)workItem;
+@end
+
+@implementation Scheduler {
+    NSMutableArray<NSTimer *> *timers;
+}
+
+- (instancetype)init
+{
+    if (self = [super init])
+        timers = [NSMutableArray new];
+    return self;
+}
+
+- (void)schedule:(WGPUWorkItem)workItem
+{
+    [timers addObject:[NSTimer scheduledTimerWithTimeInterval:0 repeats:NO block:^(NSTimer *timer) {
```

Comment:
I'm just curious why you used timers instead of GCD?

Reply:
Oh, it's just a bit more user-friendly.